src/bespokelabs/curator/request_processor/online_request_processor.py

```diff
@@ -0,0 +1,312 @@
+
+                    if resume and generic_request.original_row_idx in completed_request_ids:
+                        status_tracker.num_tasks_already_completed += 1
+                        continue
+
+                    request = APIRequest(
+                        task_id=status_tracker.num_tasks_started,
+                        generic_request=generic_request,
+                        api_specific_request=self.create_api_specific_request(generic_request),
+                        attempts_left=max_attempts,
+                        prompt_formatter=self.prompt_formatter,
+                    )
+
+                    token_estimate = self.estimate_total_tokens(request.generic_request.messages)
+
+                    # Wait for capacity if needed
+                    while not status_tracker.has_capacity(token_estimate):
+                        await asyncio.sleep(0.1)
+
+                    # Consume capacity before making request
+                    status_tracker.consume_capacity(token_estimate)
+
+                    task = asyncio.create_task(
+                        self.process_single_request(
+                            request=request,
+                            session=session,
+                            retry_queue=queue_of_requests_to_retry,
+                            save_filepath=save_filepath,
+                            status_tracker=status_tracker,
+                        )
+                    )
+                    pending_requests.append(task)
+
+                    status_tracker.num_tasks_started += 1
+                    status_tracker.num_tasks_in_progress += 1
+
+                    # Process any retries that are in the queue
+                    while not queue_of_requests_to_retry.empty():
+                        retry_request = await queue_of_requests_to_retry.get()
+                        token_estimate = self.estimate_total_tokens(
+                            retry_request.generic_request.messages
+                        )
+
+                        # Wait for capacity if needed
+                        while not status_tracker.has_capacity(token_estimate):
+                            await asyncio.sleep(0.1)
+
+                        # Consume capacity before making request
+                        status_tracker.consume_capacity(token_estimate)
+
+                        task = asyncio.create_task(
+                            self.process_single_request(
+                                request=retry_request,
+                                session=session,
+                                retry_queue=queue_of_requests_to_retry,
+                                save_filepath=save_filepath,
+                                status_tracker=status_tracker,
+                            )
+                        )
+                        pending_requests.append(task)
+
+            # Wait for all tasks to complete
```

Comment:
old code has a cool down if a rate limit error in encounter (local accounting messed up) - should part of this base class
https://github.com/bespokelabsai/curator/blob/main/src/bespokelabs/curator/request_processor/openai_online_request_processor.py#L417-L432